x-pack/plugins/apm/typings/es_schemas/Stackframe.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+interface IStackframeBase {
+  filename: string;
+  line: {
+    number: number;
+    column?: number;
+    context?: string;
+  };
+  abs_path?: string;
+  colno?: number;
+  context_line?: string;
+  function?: string;
+  library_frame?: boolean;
+  exclude_from_grouping?: boolean;
+  module?: string;
+  context?: {
+    post?: string[];
+    pre?: string[];
+  };
+  sourcemap?: {
+    updated?: boolean;
+    error?: string;
+  };
+  vars?: {
+    [key: string]: unknown;
+  };
+  orig?: {
+    filename?: string;
+    abs_path?: string;
+    function?: string;
+    lineno?: number;
+    colno?: number;
+  };
+}
+
+interface IStackframeWithoutLineContext extends IStackframeBase {
+  line: {
+    number: number;
+    column?: number;
+    context: undefined;
+  };
+}
+
+export interface IStackframeWithLineContext extends IStackframeBase {
+  line: {
+    number: number;
+    column?: number;
+    context: string;
+  };
+  context: {
+    post: string[];
+    pre: string[];
+  };
+}
+
+export type IStackframe =
+  | IStackframeWithoutLineContext
+  | IStackframeWithLineContext;
```

Comment:
This is a bit interesting. I want to have two versions of the same base type:
One that contains `stackframe.line.context` and another one that doesn't.

Reply:
Seems like you might be able to flip it around too, and make the base not mention `line.context`, then extend from Base and have line specify context (along with the other keys in the Base line, as TS will require): 

Example: https://www.typescriptlang.org/play/#src=interface%20Base%20%7B%0D%0A%20%20%20%20name%3A%20string%3B%0D%0A%20%20%20%20line%3A%20%7B%0D%0A%20%20%20%20%20%20%20%20number%3A%20number%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Ainterface%20WithLineContext%20extends%20Base%20%7B%0D%0A%20%20%20%20line%3A%20%7B%0D%0A%20%20%20%20%20%20%20%20number%3A%20number%3B%0D%0A%20%20%20%20%20%20%20%20context%3A%20string%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Afunction%20a(val%3A%20WithLineContext)%20%7B%0D%0A%20%20%20%20return%20val.line.context%3B%0D%0A%7D%0D%0A%0D%0Afunction%20b(val%3A%20Base)%20%7B%0D%0A%20%20%20%20return%20val.line.number%3B%0D%0A%7D